services/ux-backend/handlers/onboardingtokens/handler.go

```diff
@@ -33,31 +35,42 @@ func HandleMessage(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours
 
 func handlePost(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
 	var storageQuotaInGiB *uint
+	// to ensure backward compatibility, if the body is empty, we assume it to be client
+	role := services.ClientRole
 	// When ContentLength is 0 that means request body is empty and
 	// storage quota is unlimited
 	var err error
 	if r.ContentLength != 0 {
-		var quota = struct {
+		var body = struct {
+			Role  string `json:"role"`
 			Value uint   `json:"value"`
 			Unit  string `json:"unit"`
 		}{}
-		if err = json.NewDecoder(r.Body).Decode(&quota); err != nil {
+		if err = json.NewDecoder(r.Body).Decode(&body); err != nil {
 			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
-
-		if quota.Value == 0 || quota.Value > math.MaxInt {
-			http.Error(w, fmt.Sprintf("invalid value sent in request body, value should be greater than 0 and less than %v: %v", math.MaxInt, quota.Value), http.StatusBadRequest)
-			return
-		}
-		unitAsGiB, ok := unitToGib[quota.Unit]
-		if !ok {
-			http.Error(w, fmt.Sprintf("invalid Unit type sent in request body, Valid types are [Gi,Ti,Pi]: %v", quota.Unit), http.StatusBadRequest)
+		if strings.ToLower(body.Role) == string(services.MirroringPeerRole) {
+			role = services.MirroringPeerRole
+		} else if strings.ToLower(body.Role) == string(services.ClientRole) || body.Role == "" {
+			// to ensure backward compatibility, if the role is empty, we assume it to be client
+			role = services.ClientRole
+			if body.Value == 0 || body.Value > math.MaxInt {
+				http.Error(w, fmt.Sprintf("invalid value sent in request body, value should be greater than 0 and less than %v: %v", math.MaxInt, body.Value), http.StatusBadRequest)
+				return
+			}
+			unitAsGiB, ok := unitToGib[body.Unit]
+			if !ok {
+				http.Error(w, fmt.Sprintf("invalid Unit type sent in request body, Valid types are [Gi,Ti,Pi]: %v", body.Unit), http.StatusBadRequest)
+				return
+			}
+			storageQuotaInGiB = ptr.To(unitAsGiB * body.Value)
+		} else {
+			http.Error(w, fmt.Sprintf("invalid Role sent in request body, Valid roles are [mirroring-peer,client]: %v", body.Role), http.StatusBadRequest)
```

Comment:
We should not do that. This will allow anyone to generate a peer onboarding token.